src/connection.c

```diff
@@ -330,3 +328,24 @@ ConnectionGetOptionValue(PGconn *connection, char *optionKeyword)
 
 	return optionValue;
 }
+
+ShardId DoForAllShards(List* shardConnections, ShardAction action, void* arg)
```

Comment:
You really think that duplicating the same code (two nested loops) in three places is really good idea?
What is confusing with DoForAllShards? It is standard map function...